querqy-for-lucene/querqy-solr/src/main/java/querqy/solr/ClassicRewriteChainLoader.java

```diff
@@ -0,0 +1,115 @@
+
+    private static final Logger LOG = LoggerFactory.getLogger(TermQueryCachePreloader.class);
+
+    private String rewriterRequestHandlerName;
+
+    public ClassicRewriteChainLoader(final SolrCore core) {
+        super(core);
+    }
+
+    @Override
+    public void init(final NamedList args) {
+        super.init(args);
+
+        final String rewriteHandlerName = (String) getArgs().get(CONF_REWRITER_REQUEST_HANDLER);
+        rewriterRequestHandlerName = !isNullOrEmpty(rewriteHandlerName) ? rewriteHandlerName : DEFAULT_HANDLER_NAME;
+    }
+
+    @Override
+    public void newSearcher(final SolrIndexSearcher newSearcher, final SolrIndexSearcher currentSearcher) {
+        if (currentSearcher == null) {
+            loadRewriteChain(newSearcher.getCore());
+        }
+    }
+
+    @VisibleForTesting
+    String getRewriterRequestHandler() {
+        return this.rewriterRequestHandlerName;
+    }
+
+    private void loadRewriteChain(final SolrCore core) {
+
+        final NamedList<?> chainConfig = (NamedList<?>) getArgs().get("rewriteChain");
+        if (chainConfig != null) {
+
+            final GZIPAwareResourceLoader resourceLoader = new GZIPAwareResourceLoader(core.getResourceLoader());
+            final QuerqyRewriterRequestHandler requestHandler = (QuerqyRewriterRequestHandler) core.getRequestHandler(rewriterRequestHandlerName);
+
+            @SuppressWarnings("unchecked") final List<NamedList<?>> rewriterConfigs = (List<NamedList<?>>) chainConfig.getAll("rewriter");
+            if (rewriterConfigs != null) {
+
+                final Set<String> seenRewriterIds = new HashSet<>(rewriterConfigs.size());
+
+                for (NamedList<?> config : rewriterConfigs) {
+
+                    final String id = (String) config.get("id");
+                    if (id == null) {
+                        throw new IllegalArgumentException("Rewriter missing id field!");
+                    }
+
+                    if (!seenRewriterIds.add(id)) {
+                        throw new RuntimeException("Rewriter id: " + id + "already defined.");
+                    }
+
+                    try {
+                        final ModifiableSolrParams params = new ModifiableSolrParams();
+                        params.set(PARAM_ACTION, SAVE.name());
+
+                        final String className = (String) config.get("class");
+
+                        final Map<String, Object> jsonBody = ((ClassicConfigurationParser) SolrRewriterFactoryAdapter
+                                .loadInstance(id, className))
+                                .parseConfigurationToRequestHandlerBody((NamedList<Object>) config, resourceLoader);
+
+                        final SolrQueryRequestBase req = new SolrQueryRequestBase(core, params) {
+                        };
+                        req.setContentStreams(newArrayList(new ContentStreamBase.StringStream(JsonUtil.toJson(jsonBody), UTF_8.name())));
```

Comment:
newArrayList - this list can be immutable. We can simply use https://docs.oracle.com/javase/6/docs/api/java/util/Collections.html#singletonList(T) and get rid of the Guava dependency at the same go.